.travis.yml

```diff
@@ -16,6 +16,38 @@ script:
 # Docker Run to Execute Unit E2E Selenium Package
 - echo "Running E2E Tests"
 - docker run -e APP_KEY -e APP_TOKEN -e board_id -e list_id_Done -e list_id_Pending -e list_id_ToDo -e SECRET_KEY todo-app:test Test/selenium_test.py
+- echo "All Tests Have Been Completed"
+
+# DockerHubLogin
+- docker login --username "$docker_login" --password "$docker_pass"
+
+# Docker Build the Prod and DockerPush
+- docker build --target production --tag todo-app:prod .
+- docker tag todo-app:prod legendarypan/todo-app:prod
+- docker push legendarypan/todo-app:prod
+
+# Pre-deploy Install
+- curl -sS https://cli-assets.heroku.com/install.sh | sh
+# - heroku update
+# - heroku plugins:install @heroku-cli/plugin-container-registry
+
+
+# using heroku credentials
+# NOTES use auth token instead of API key and move to environment variable like docker_login 
+- docker login --username=$HEROKU_LOGIN --password=$HEROKU_API_KEY registry.heroku.com
+- docker tag legendarypan/todo-app:prod registry.heroku.com/fengs-todo-app/web
+- docker push registry.heroku.com/fengs-todo-app/web
```

Comment:
Since this section is tied to the deployment and you only want to do it when you're going to deploy, you could put it in a `before_deploy` stage rather than at the end of `script`.